packages/@lwc/engine-core/src/renderer.ts

```diff
@@ -10,267 +13,75 @@ export type HostElement = any;
-    getChildren = getChildrenImpl;
-}
-
-type getChildNodesFunc = (element: E) => NodeList;
-export let getChildNodes: getChildNodesFunc;
-export function setGetChildNodes(getChildNodesImpl: getChildNodesFunc) {
-    getChildNodes = getChildNodesImpl;
-}
-
-type getFirstChildFunc = (element: E) => N | null;
-export let getFirstChild: getFirstChildFunc;
-export function setGetFirstChild(getFirstChildImpl: getFirstChildFunc) {
-    getFirstChild = getFirstChildImpl;
-}
-
-type getFirstElementChildFunc = (element: E) => E | null;
-export let getFirstElementChild: getFirstElementChildFunc;
-export function setGetFirstElementChild(getFirstElementChildImpl: getFirstElementChildFunc) {
-    getFirstElementChild = getFirstElementChildImpl;
-}
-
-type getLastChildFunc = (element: E) => N | null;
-export let getLastChild: getLastChildFunc;
-export function setGetLastChild(getLastChildImpl: getLastChildFunc) {
-    getLastChild = getLastChildImpl;
-}
-
-type getLastElementChildFunc = (element: E) => E | null;
-export let getLastElementChild: getLastElementChildFunc;
-export function setGetLastElementChild(getLastElementChildImpl: getLastElementChildFunc) {
-    getLastElementChild = getLastElementChildImpl;
-}
-
-type isConnectedFunc = (node: N) => boolean;
-export let isConnected: isConnectedFunc;
-export function setIsConnected(isConnectedImpl: isConnectedFunc) {
-    isConnected = isConnectedImpl;
-}
-type insertStylesheetFunc = (content: string, target?: ShadowRoot) => void;
-export let insertStylesheet: insertStylesheetFunc;
-export function setInsertStylesheet(insertStylesheetImpl: insertStylesheetFunc) {
-    insertStylesheet = insertStylesheetImpl;
-}
-
-type assertInstanceOfHTMLElementFunc = (elm: any, msg: string) => void;
-export let assertInstanceOfHTMLElement: assertInstanceOfHTMLElementFunc;
-export function setAssertInstanceOfHTMLElement(
-    assertInstanceOfHTMLElementImpl: assertInstanceOfHTMLElementFunc
-) {
-    assertInstanceOfHTMLElement = assertInstanceOfHTMLElementImpl;
-}
-
-type defineCustomElementFunc = (
-    name: string,
-    constructor: CustomElementConstructor,
-    options?: ElementDefinitionOptions
-) => void;
-export let defineCustomElement: defineCustomElementFunc;
-export function setDefineCustomElement(defineCustomElementImpl: defineCustomElementFunc) {
-    defineCustomElement = defineCustomElementImpl;
-}
-
-type getCustomElementFunc = (name: string) => CustomElementConstructor | undefined;
-export let getCustomElement: getCustomElementFunc;
-export function setGetCustomElement(getCustomElementImpl: getCustomElementFunc) {
-    getCustomElement = getCustomElementImpl;
+let defaultRenderer: RendererAPI;
+
+export function setDefaultRenderer(renderer: RendererAPI) {
+    if (defaultRenderer !== undefined) {
+        throw Error(`Invalid Renderer Initialization`);
+    }
+    defaultRenderer = renderer;
+}
+
+export function getRendererFromVNode(vnode: VBaseElement): RendererAPI {
```

Comment:
In most cases, the expression `vnode.data.renderer` would be evaluated as `undefined` and would return the fallback value `defaultRender`.  

This PR sprinkles `getRendererFromVNode` throughout `render.ts`, `hydrate.ts`, and `modules/*.ts`. Since the `renderer` property doesn't change once being set, it's something we retrieve once per VNode and pass it around as an argument. In theory, we should only invoke this method in: `mountElement`, `patchElement`, `mountCustomElement`, `pathCustomElement`. This would avoid unnecessary lookups and improve the overall performance of patching.

Reply:
we can chat about it!

Reply:
ok, I went ahead and made this change... seems like we have reduced the lookup significant.